Add Identity Verification product support

punkisdead opened the pull request:
## Summary

This PR adds support for Plaid's Identity Verification product to the library.

## Changes

- **New module**: `Plaid.IdentityVerification` with complete API support
  - `create/2` - Create a new identity verification
  - `get/2` - Retrieve verification details  
  - `list/2` - List verifications with pagination
  - `retry/2` - Retry a verification attempt

- **Data structures**: Comprehensive structs for all verification response types including:
  - Template, User (Name, Address, IDNumber)
  - Steps with all verification stages
  - DocumentaryVerification with document analysis
  - SelfieCheck with facial verification
  - KYCCheck, RiskCheck, WatchlistScreening
  - ListResponse for pagination

- **Test coverage**: Full test suite with 13 tests
  - Unit tests using Mox for behavior mocking
  - Integration tests using Bypass for HTTP simulation
  - Both success and error scenarios covered

- **Test fixtures**: Added factory functions for identity verification responses

## Implementation Notes

- Follows all existing library patterns and conventions
- Uses the same testing approach as other products
- All structs derive Jason.Encoder for serialization
- Complete type specifications for Dialyzer
- Documentation links to official Plaid API docs

## Testing

All 157 tests pass including the 13 new identity verification tests.

punkisdead committed:
Add support for Identity Verification product

Implements the complete Identity Verification API with:
- Four endpoints: create, get, list, and retry
- Comprehensive data structures for all verification types
- Full test coverage with unit and integration tests
- Follows existing library patterns and conventions

wfgilman commented on lib/plaid/identity_verification.ex:
```diff
@@ -0,0 +1,589 @@
+            }
+    end
+
+    defmodule Address do
```
Is this meaningfully different from `Plaid.Accounts.Account.Owner.Address` which is just a map? Thoughts on standardizing address across the two modules?

punkisdead replied:
Plaid Identity/Accounts Owner Address:
  ```
%{
    data: %{
      street: "564 Main Street, APT 15",
      city: "Malakoff",
      region: "NY",
      postal_code: "14236",
      country: "US"
    },
    primary: true
  }
```
  - Uses a data map wrapper (all fields nullable)
  - Has primary boolean for multi-address support
  - Single street field includes apartment/suite
  - Reflects data from financial institutions (flexible, sometimes incomplete)

  Plaid Identity Verification User Address:
  ```
%{
    street: "123 Main St",
    street2: "Apt 15",  # Separate field!
    city: "San Francisco",
    region: "CA",
    postal_code: "94108",
    country: "US"
  }
```
  - Direct fields (no wrapper)
  - No primary flag (single address concept)
  - Separate street2 for apartment/suite numbers
  - Some fields required (user-provided data for verification)

punkisdead replied:
They are similar, but not exact.

wfgilman commented on lib/plaid/identity_verification.ex:
```diff
@@ -0,0 +1,589 @@
+    @moduledoc """
+    Plaid Identity Verification Steps data structure.
+
+    Each step will be one of the following values: `active`, `success`, `failed`,
```
FMI, which field in the struct has this value? Asking to see whether it makes sense to make them explicit `@type` values, perhaps as atoms

punkisdead replied:
the steps object in the API response looks like the following:
```
"steps": {
		"accept_tos": "success",
		"documentary_verification": "failed",
		"kyc_check": "failed",
		"risk_check": "failed",
		"selfie_check": "not_applicable",
		"verify_sms": "success",
		"watchlist_screening": "not_applicable"
	}
```
Each of the keys defined in the object will have one of the types above. I thought about making it an explicit type, but noticed that none of the other structs in the project enforced strict enum types. I'm ok with making it more specific, however if Plaid ever changes this, the library will need to be updated too.

wfgilman commented on lib/plaid/identity_verification.ex:
```diff
@@ -0,0 +1,589 @@
+          }
+  end
+
+  defmodule ListResponse do
```
Can you please rename this to `Verifications` or similar? The library has a precedent for having domain-named data structures instead of generic ones, e.g. [Payments](https://github.com/wfgilman/plaid-elixir/blob/47df2f4a44c7a248add368828f9cff44a4ddeb1d/lib/plaid/payment_initiation/payments.ex#L1-L21)

wfgilman commented on lib/plaid/identity_verification.ex:
```diff
@@ -0,0 +1,589 @@
+    )
+  end
+
+  defp map_list_response(body) do
```
Ideally this private function would change long with the renaming of the `ListResponse` module following the pattern used for `Payments`: https://github.com/wfgilman/plaid-elixir/blob/47df2f4a44c7a248add368828f9cff44a4ddeb1d/lib/plaid/payment_initiation/payments.ex

wfgilman reviewed (changes requested):
Thanks for the PR! Looks great - I just have a couple questions and a request to rename a response object to better follow the pattern for list responses in the library.

punkisdead committed:
Rename ListResponse to Verifications in IdentityVerification module

Renamed the ListResponse struct to Verifications for better clarity and
consistency. Also renamed the corresponding map_list_response function
to map_verifications.

wfgilman reviewed (changes requested):
Looks like there's a remaining reference to `ListResponse`

punkisdead committed:
Fix missed renaming

punkisdead commented:
> Looks like there's a remaining reference to `ListResponse`

Fixed it.
